go.mod

```diff
@@ -287,19 +289,11 @@ require (
 	gopkg.in/warnings.v0 v0.1.2 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
-	gotest.tools v2.2.0+incompatible // indirect
 	honnef.co/go/tools v0.4.3 // indirect
 	mvdan.cc/gofumpt v0.4.0 // indirect
 	mvdan.cc/interfacer v0.0.0-20180901003855-c20040233aed // indirect
 	mvdan.cc/lint v0.0.0-20170908181259-adc824a0674b // indirect
 	mvdan.cc/unparam v0.0.0-20221223090309-7455f1af531d // indirect
 )
 
-retract (
-	// a regression was introduced
-	v0.38.4
-	// a breaking change was introduced
-	v0.38.3
-	// superseeded by v0.38.3 because of ASA-2024-001
-	[v0.38.0, v0.38.2]
-)
+replace github.com/ethereum/go-ethereum => github.com/maticnetwork/bor v1.0.4
```

Comment:
Just curious, what would break if we removed this "replace" (I'm thinking here of the time when we'll upstream this)

Reply:
From code perspective, I don't see anything breaking since we wouldn't be making any changes to Geth's crypto library in our fork (i.e [Bor](https://github.com/maticnetwork/bor/tree/develop)). From a dependency chain perspective I'm not sure since heimdall-v2 would need bor and Comet would use Geth. WDYT @marcello33 ?

Reply:
@sergio-mena Are you going to use go-ethereum implementation for you uncompressed public key case in future?

Reply:
@VAIBHAVJINDAL3012 we will need to discuss this when upstreaming the uncompressed changes on CometBFT (we haven't started yet), but my current view is that we'd probably have to depend on go-ethereum, unless we find a "better" (to be defined what "better means here) library that contains an implementation  compatible with  go-ethereum.

Reply:
I found the only occurrence of `github.com/ethereum/go-ethereum` w.r.t code in `crypto/secp256k1/secp256k1.go` where we are leveraging the `crypto` package for hashing and signing purposes, the underlying implementation of which will not differ in our fork (i.e in `bor`).
Hence I think we don't need to replace with bor in the dependency.

Reply:
Excellent, thanks for looking into it!

Reply:
Resolved (AFAIAC)